Introducing Miniflux service to DebOps

BerkhanBerkdemir opened the pull request:
Miniflux is a minimalist and opinionated feed reader. It is written in
Golang which is distributed as a single binary. This commit provides a
playbook and role to run Miniflux service on Debian and Ubuntu
distributions.

Even though this is the first commit of the playbook/role, it is not
ready for production use.

For more information about Miniflux, please check http://miniflux.app/
website.

BerkhanBerkdemir committed:
Introducing Miniflux service to DebOps

Miniflux is a minimalist and opinionated feed reader. It is written in
Golang which is distributed as a single binary. This commit provides a
playbook and role to run Miniflux service on Debian and Ubuntu
distributions.

Even though this is the first commit of the playbook/role, it is not
ready for production use.

For more information about Miniflux, please check http://miniflux.app/
website.

BerkhanBerkdemir committed:
Create symlink for playbooks/app directory

BerkhanBerkdemir committed:
Remove TODOs and miniflux__additional_groups

BerkhanBerkdemir committed:
Fix wrongly indented `collections` YAML key

BerkhanBerkdemir committed:
Emphasize documentation and use dependency keyword

BerkhanBerkdemir committed:
Add variables to manage Git installation

BerkhanBerkdemir commented:
Previously, I sketched some things to-do before I accept as this work as
done.

- [x] Allow user to install from Git
    @ `miniflux/defaults/main.yml#230`
- [ ] Add service restart handler
    @ `miniflux/defaults/main.yml#237` (?)
- [x] Use miniflux__gecos instead acording to `ansible.builtin.user`
    @ `miniflux/defaults/main.yml#279`
- [ ] Use community.general.ini_file to generate for conf. file
    @ `miniflux/tasks/main.yml#43`
- [x] Daemon should see environment variables
    @ `miniflux/tasks/main.yml#47`
- [ ] Create miniflux__database_url to manage connection string
    @ `miniflux/tasks/main.yml#47`
- [x] Manage Miniflux binary, so that can be used in systemd unit file
- [ ] Can I make PostgreSQL as Miniflux dependency? What happens if
    database is seperate?
- [x] Variables are not in use
    - `miniflux__upstream_gpg_key`
    - `miniflux__upstream_upgrade`

drybjed commented on ansible/roles/miniflux/defaults/main.yml:
```diff
@@ -0,0 +1,307 @@
+      - src: '{{ miniflux__upstream_url_mirror + "download/2.0.35/miniflux-" + miniflux__upstream_platform }}'
+        dest: 'releases/{{ miniflux__upstream_platform }}/miniflux/miniflux/2.0.35/miniflux_2.0.35_linux_amd64'
+    url_binaries:
+      - src: 'releases/{{ miniflux__upstream_platform }}/miniflux/miniflux/2.0.35/miniflux_2.0.35_linux_amd64'
```
The version could be specified in a separate variable, `miniflux__upstream_version`, this could simplify upgrades. You can also add a watch file (you can find and example in the `docker_registry` role) and when you run the `make versions` command, the script will find out if upstream released a new version.

drybjed commented on ansible/roles/miniflux/defaults/main.yml:
```diff
@@ -0,0 +1,307 @@
+#
+miniflux__comment: 'Miniflux'
+miniflux__bind: '127.0.0.1'
+miniflux__port: '3366'
```
Looking forward to it. :-)

BerkhanBerkdemir replied:
Default Miniflux configuration uses 127.0.0.1:8080. Do I really need to set some variables for `__bind` and `__port`? I was thinking maybe I can use UNIX socket for nginx and Apache?

drybjed replied:
You can use a socket, but it would be nice to have it configurable so that if somebody want to do a distributed setup, they still can. If one variable is enough to define that socket/port connection then you can go with that; otherwise if Miniflux uses bind address and port separately, then can be separate.

drybjed commented on ansible/roles/miniflux/tasks/main.yml:
```diff
@@ -0,0 +1,55 @@
+- name: Run migrations
+  # environment:
+  #   DATABASE_URL: 'postgres://{{ miniflux__database_user }}:{{ miniflux__database_password }}@{{ miniflux__database_host }}:{{ miniflux__database_port }}/{{ miniflux__database_name }}'
+  command: '/usr/local/bin/miniflux -migrate'
```
Is this command idempotent? Expect that the role can be executed multiple times against the host.

drybjed reviewed (changes requested):
Nice work so far. :+1: You should add role documentation before it will be merged into DebOps though. Keep it up! :-)

BerkhanBerkdemir committed:
Configure upstream version and use uscan

BerkhanBerkdemir committed:
Move migration configuration to conf

BerkhanBerkdemir committed:
Use unused variable for GPG

BerkhanBerkdemir committed:
Remove currently unused variable

BerkhanBerkdemir committed:
Add Miniflux role for testing in CI

BerkhanBerkdemir commented:
I would like to point out what I did so far. As @drybjed pointed out, I created a `watch-miniflux` and moved migration commend to a configuration file in 9ccbe2f8a1e9b3081c166ecc9fc208178f9930a6. I used one unused variable and removed the other. I also wanted to add initial work of CI testing in GitLab.

BerkhanBerkdemir committed:
Configure Miniflux bind & port address with GECOS

BerkhanBerkdemir committed:
Apply GitHub specific suggestion from man:uscan(1)

BerkhanBerkdemir commented:
It's been long time no commit I made to this particular feature. The only commits or changes I have so far related to documentation. I can't promise if I can push those changes in the near future. If anyone willing to write documentation and/or add variables, please let me know so that I can share raw documentation I have.

drybjed commented:
@BerkhanBerkdemir I plan to finish your PR and write documentation in the near future, so sure - show me what you've got. :-)
